core/site.go

```diff
@@ -412,8 +415,19 @@ func (site *Site) updateMeters() error {
 			soc, err := meter.(api.Battery).Soc()
 
 			if err == nil {
-				site.batterySoc += soc
+				var remainingCapacity float64
+				remainingCapacity = soc
 				site.log.DEBUG.Printf("battery %d soc: %.0f%%", i+1, soc)
+				if capacity, ok := meter.(api.BatteryCapacity); ok {
+					remainingCapacity *= capacity.Capacity()
+					sumBatteryCapacity += capacity.Capacity()
+				} else {
+					// We add 1 here, since the battery will be counted with a
+					// capacity factor of 1
+					sumBatteryCapacity += 1
```

Comment:
Der Zweig kann sogar weg. Wenn nach dem Loop noch 0 raus kommt einfach durch len(batterymeters) dividieren so wie das vorher auch war?

Reply:
Kommt ja nicht unbedingt 0 bei raus.
Kann ja sein, dass es eine Mischung gibt, bei der manche Batterien eine Kapazität haben und andere nicht.
Und berechnet wird dann soc*1 für die aktuelle Kapazität, daher muss man auch die Gesamtkapazität +1 rechnen.
Und wenn keine Batterie eine Kapazität gesetzt hat, dann ergibt sich so exakt die Rechnung von vorher, da dann `sumBatteryCapacity = len(batteryMeters)` 
Ist einfach und für alle Fälle korrekt.

Reply:
> Kann ja sein, dass es eine Mischung gibt, bei der manche Batterien eine Kapazität haben und andere nicht.

In dem Fall ist das Ergebnis immer und einfach nur anders falsch.

Reply:
Stimmt. Aber für die beiden "großen" Fälle, "alle oder keine haben eine Kapazität" ist das so ebenfalls richtig. Und wir würden den else Zweig hier durch eine neue Abfrage und komplett andere Berechnung weiter unten ersetzen und zumindest mathematisch für den gemischten Fall auch noch falsch rechnen.
Halte ich nicht für besser, um ehrlich zusein.

Reply:
Ich hab mal beides so umgebaut wie's mir gefällt- damit ist die Kapa jetzt auch auf mqtt sichtbar. Was meinst Du?

Reply:
Mom, das ist natürlich falsch so

Reply:
Done. Hab auch gleich noch das error logging in dem Block angehübscht